server/sonar-server/src/test/java/org/sonar/server/ui/ws/ComponentActionTest.java

```diff
@@ -219,6 +219,50 @@ public void return_component_info_when_snapshot() throws Exception {
   }
 
   @Test
+  public void return_component_info_when_file_on_master() throws Exception {
+    init();
+    OrganizationDto organization = dbTester.organizations().insertForKey("my-org2");
+    ComponentDto main = componentDbTester.insertMainBranch(organization, p -> p.setName("Sample"), p -> p.setDbKey("sample"));
+    userSession.addProjectPermission(UserRole.USER, main);
+
+    ComponentDto dirDto = newDirectory(main, "src");
+    componentDbTester.insertComponent(dirDto);
+
+    ComponentDto fileDto = newFileDto(main, dirDto)
+      .setUuid("abcd")
+      .setName("Main.xoo")
+      .setDbKey("sample:src/Main.xoo");
+    componentDbTester.insertComponent(fileDto);
```

Comment:
You can inline fileDto

Reply:
I use `fileDto.getDbKey()` a few lines below this. Sure I could cram all the `componentDbTester.insertComponent(...).getDbKey()` in there, but I think readability would suffer.

Reply:
Same as before, I would write : 
'ComponentDto fileDto = componentDbTester.insertComponent(newFileDto(main, dirDto).set...)'